### PR TITLE
JAMES-2589 Separate injection for choosing blob store configuration

### PR DESCRIPTION
**Pull request by chibenwa**
This enables a easy inject in tests

### CODE REVIEW

**Comment by trantienduchn** [comment]
could you change `objectstore` file name to `objectstorage` file name in `BlobStoreChoosingModule` ?

**Comment by trantienduchn** [comment]
> could you help me change objectstore file name to objectstorage file name in BlobStoreChoosingModule ?

I will do this

File: server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageDependenciesModule.java [hunk 1]
```diff
@@ -44,7 +44,7 @@
 
 public class ObjectStorageDependenciesModule extends AbstractModule {
 
-    private static final String OBJECTSTORAGE_CONFIGURATION_NAME = "objectstorage";
+    public static final String OBJECTSTORAGE_CONFIGURATION_NAME = "objectstorage";
```
Comment (mbaechler) [inline server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageDependenciesModule.java]: this file is related to swift/S3, you can reuse it elsewhere.
If you want the same file to be used for several things, you have to use a name declared elsewhere

File: server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/objectstore/BlobStoreChoosingModule.java [hunk 2]
```diff
@@ -60,10 +59,10 @@ protected void configure() {
     @Singleton
     BlobStoreChoosingConfiguration provideChoosingConfiguration(PropertiesProvider propertiesProvider) throws ConfigurationException {
         try {
-            Configuration configuration = propertiesProvider.getConfiguration(BLOBSTORE_CONFIGURATION_NAME);
+            Configuration configuration = propertiesProvider.getConfiguration(OBJECTSTORAGE_CONFIGURATION_NAME);
```
Comment (mbaechler) [inline server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/objectstore/BlobStoreChoosingModule.java]: here we should only know about blobstore, not objectstorage (that is an implementation)

File: server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageDependenciesModule.java (hunk 1 above)
Comment (trantienduchn) [reply server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageDependenciesModule.java]: I remember, we had a discussion and came to the choice of using only one configuration file for whole things related to blobstore. That means, put all configuration on one  file.

May be the file name is not really good. But using one file is final decision. You was not there in the discussion.

Comment (chibenwa) [reply server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageDependenciesModule.java]: Let's do `blobstore.properties` + `objectstorage.properties`

**Comment by chibenwa** [comment]
(Was :green: )

File: dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/blobstore.properties [hunk 3]
```diff
@@ -1,3 +1,5 @@
+objectstore.implementation=swift
```
Comment (mbaechler) [inline dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/blobstore.properties]: oh, well, it should be objectstorage and not swift, I missed that previously.
objectstorage is supposed to also handle S3 (see objectorage.provider)

Comment (mbaechler) [inline dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/blobstore.properties]: and what's the point of prefixing by objectstore ? `implementation`  or `backend` should be enough

File: server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/objectstore/BlobStoreChoosingConfiguration.java [hunk 4]
```diff
@@ -72,6 +73,14 @@ static BlobStoreChoosingConfiguration from(Configuration configuration) {
         return new BlobStoreChoosingConfiguration(blobStoreImplName);
     }
 
+    public static BlobStoreChoosingConfiguration cassandra() {
+        return new BlobStoreChoosingConfiguration(BlobStoreImplName.CASSANDRA);
+    }
+
+    public static BlobStoreChoosingConfiguration swift() {
```
Comment (mbaechler) [inline server/container/guice/cassandra-rabbitmq-guice/src/main/java/org/apache/james/modules/objectstore/BlobStoreChoosingConfiguration.java]: should be objectstorage

File: dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/blobstore.properties [hunk 5]
```diff
@@ -1,4 +1,4 @@
-objectstore.implementation=swift
+implementation=object-storage
```
Comment (mbaechler) [inline dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/blobstore.properties]: it's not consistent with "objectstore" just below

Comment (chibenwa) [reply dockerfiles/run/guice/cassandra-rabbitmq/destination/conf/blobstore.properties]: > it's not consistent with "objectstore" just below

Looking at the final diff, I do not see any mention to `objectstore`.

Could you develop?

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**